### PR TITLE
ENG-894: [PS-8.0] Improve autotests

### PR DESCRIPTION
**Pull request by Sudokamikaze**

### CODE REVIEW

File: build-ps/test-build.sh [hunk 1]
```diff
@@ -1,61 +1,75 @@
...
     for DIR in $DIRLIST; do
-        check_libs $DIR | tee -a $TMP_DIR/libs_err.log
+        check_libs "$DIR" >> "$TMP_DIR"/libs_err.log
     done
 
-    if [[ ! -z $(cat $TMP_DIR/libs_err.log | grep "not found") ]]; then
+    if [[ ! -z "$(cat $TMP_DIR/libs_err.log | grep "not found")" ]]; then
         echo "ERROR: There are missing libraries: "
-        cat $TMP_DIR/libs_err.log
+        cat "$TMP_DIR"/libs_err.log | grep "not found"
```
Comment (hors) [inline build-ps/test-build.sh]: grep "not found" "$TMP_DIR"/libs_err.log

File: build-ps/test-build.sh [hunk 2]
```diff
@@ -1,61 +1,75 @@
...
+    cd "${TARBALL%.tar.gz}"
 
     echo "Checking ELFs for not found"
     for DIR in $DIRLIST; do
-        check_libs $DIR | tee -a $TMP_DIR/libs_err.log
+        check_libs "$DIR" >> "$TMP_DIR"/libs_err.log
     done
 
-    if [[ ! -z $(cat $TMP_DIR/libs_err.log | grep "not found") ]]; then
+    if [[ ! -z "$(cat $TMP_DIR/libs_err.log | grep "not found")" ]]; then
```
Comment (hors) [inline build-ps/test-build.sh]: grep "not found" $TMP_DIR/libs_err.log

File: build-ps/test-build.sh [hunk 3]
```diff
@@ -1,61 +1,75 @@
...
-    tar xf $CURDIR/$TARBALL
-    cd ${TARBALL%.tar.gz}
+    cd "$TMP_DIR"
+    tar xf "$CURDIR/$TARBALL"
+    cd "${TARBALL%.tar.gz}"
 
     echo "Checking ELFs for not found"
     for DIR in $DIRLIST; do
-        check_libs $DIR | tee -a $TMP_DIR/libs_err.log
+        check_libs "$DIR" >> "$TMP_DIR"/libs_err.log
```
Comment (hors) [inline build-ps/test-build.sh]: In this case we will not have the output in standard output  only in log. Is it ok?

**Review by hors — CHANGES_REQUESTED** [review]

File: build-ps/test-build.sh (hunk 3 above)
Comment (Sudokamikaze) [reply build-ps/test-build.sh]: They will be displayed if script was launched with "bash -x"

And there's another invocation of ELFs check couple of few lines below

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
where is test for minimal tarball?
minimal tarball will not work as tarball name doesn't match directory name

File: build-ps/test-build.sh [hunk 4]
```diff
@@ -1,61 +1,85 @@
...
-    cd mysql-test
-    MTR_BUILD_THREAD=auto ./mtr \
-            --force \
-            --max-test-fail=0 \
-            main.1st 
-    MTR_BUILD_THREAD=auto ./mtr \
-            --force \
-            --suite=auth_sec
+        echo "Invoking dbdeployer to make a test run"
+        dbdeployer unpack --sandbox-binary="$TMP_DIR"/dbdeployer/tarball --prefix=ps "$CURDIR/$TARBALL"
```
Comment (EvgeniyPatlan) [inline build-ps/test-build.sh]: it will fail due to difference in tarball and directory name. please also fix directory name creation for minimal tarball

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

File: build-ps/test-build.sh [hunk 5]
```diff
@@ -1,61 +1,81 @@
...
-    tar xf $CURDIR/$TARBALL
-    cd ${TARBALL%.tar.gz}
-
-    echo "Checking ELFs for not found"
-    for DIR in $DIRLIST; do
-        check_libs $DIR | tee -a $TMP_DIR/libs_err.log
-    done
+        echo "Building ELFs ldd output list"
+        for DIR in $DIRLIST; do
+            if ! check_libs "$DIR" >> "$TMP_DIR"/libs_err.log; then
```
Comment (EvgeniyPatlan) [inline build-ps/test-build.sh:50]: please add []

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: build-ps/test-build.sh (hunk 5 above)
Comment (Sudokamikaze) [reply build-ps/test-build.sh:50]: Expected comparison operator (don't wrap commands in []/[[]]). shellcheck(SC1072)

**Review by EvgeniyPatlan — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Review by vorsel — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**